src/Neo/SmartContract/Native/NeoToken.cs

```diff
@@ -327,13 +327,35 @@ public BigInteger UnclaimedGas(DataCache snapshot, UInt160 account, uint end)
             return CalculateBonus(snapshot, state, end);
         }
 
+        [ContractMethod(Hardfork.HF_Echidna, RequiredCallFlags = CallFlags.States | CallFlags.AllowNotify)]
+        private async ContractTask OnNEP17Payment(ApplicationEngine engine, UInt160 from, BigInteger amount, StackItem data)
+        {
+            if (engine.CallingScriptHash != GAS.Hash)
+                throw new InvalidOperationException("only GAS is accepted");
+
+            if ((long)amount != GetRegisterPrice(engine.SnapshotCache))
+                throw new ArgumentException("incorrect GAS amount for registration");
+
+            var pubkey = ECPoint.DecodePoint(data.GetSpan(), ECCurve.Secp256r1);
+
+            if (!RegisterInternal(engine, pubkey))
+                throw new InvalidOperationException("failed to register candidate");
```

Comment:
We don't burn it before throw?

Reply:
a reason why a want to update these exception name to uncatchableexception~~~

Reply:
`Uncatchable` can also be misunderstood, I can catch it in various places, it's just that VM/contracts are not even aware of this happening.

Reply:
uncatachable should be named as normal. Name should be `VMException` with inner exception of normal name. Thats kind of how it done currently in the vm.